lib/tree.js

```diff
@@ -12,40 +12,115 @@ class Tree {
         this.root = null;
     }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+    // Time Complexity: O(log n)
+    // Space Complexity: O(log n)
     add(key, value) {
```

Comment:
👍 The space/time complexity is O(log n) if the tree is balanced and O(n) if it's not.